vm/systemSmartContracts/eei.go

```diff
@@ -451,6 +542,7 @@ func (host *vmContext) ExecuteOnDestContext(destination []byte, sender []byte, v
 	} else {
 		// all changes must be deleted
 		host.outputAccounts = make(map[string]*vmcommon.OutputAccount)
+		host.storageUpdate = currContext.storageUpdate
```

Comment:
is this ok? Does not match the comment. We actually preserve the storage updates here